engine/servermaster/http.go

```diff
@@ -86,3 +98,68 @@ func shouldForwardJobAPI(r *http.Request) bool {
 	}
 	return true
 }
+
+// JobHTTPClient is the http client for job operations, like 'get job detail'
+type JobHTTPClient interface {
```

Comment:
I think `JobHTTPClient` should be put in a different package. The package `servermaster` already has too many responsibilities.

Reply:
Move to a pkg `httputil`.